### ISSUE TITLE
g502 media buttons rarely work, g610 buttons work perfectly

### ISSUE THREAD

**Issue by aahrg** [MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-#2531]
I have set the buttons on my g502 mouse to various media controls (play/pause, next/previous track), and they only seem to work 1 in 5 times I press them. My g610 keyboard has hardware media keys, and they work every time. I saw someone in another thread was told to use LLkeyboardtest to verify that their logitech input device was sending the correct keycodes (Mine are, 179, 176, 177 for play/pause and next/previous track on both the mouse and keyboard.) and while trying this program, it seems that the mouse buttons are much more successful in making the player respond with LLkeyboardtest open (doesn't have to be in focus, just running). I upgraded my Logitech Gaming Software to the latest version, which didn't change anything. I am not running any kind of media player extension in chrome, all other media player applications are closed, and I've also tried disabling all my chrome extensions as well as exiting everything but LGS in my system tray.

These media buttons worked fine in spotify, as well as in an older version of this program way back before I switched to spotify.

Windows 10 pro 64 bit version 1511

Steps to reproduce: 
1. set g502 buttons to various media controls in logitech gaming software v8.92.67 (or whichever version I had before upgrading while troubleshooting)
2. attempt to use media buttons
3. get frustrated and just use the ones on keyboard

**Comment by davidsenk** [comment]
I can second that this drives me insane, the media controls on my bluetooth headsphones, and on my keyboard, work just fine everywhere (including GPMDP), but the ones I setup on my Logitech mouse work everywhere except for GPMDP. 

It drives me crazy because I'm so used to using the mouse controls, which work great in every other media player, and they do nothing in GPMDP.

**Comment by aahrg** [comment]
fixed it on my side by changing to onboard profiles rather than game detection. Good enough for me since I never actually used game profiles, but those who do would still require a fix

**Comment by beardofbeespool** [comment]
@aahrg can you elaborate on what you did?

**Comment by aahrg** [comment]
The g502 mouse has onboard memory for 3 macro profiles which you can switch between using a button on the mouse, or you can have the driver detect if specific games are running and have it load a profile specific to that game, with a default profile for when you're not gaming. I had set up media controls on the default profile so that I could control my music from the mouse.

When I encountered the issue in my OP, I was using the game detection mode so that I could also bind in game functions to my mouse. However, I found that when I tried using linux (to troubleshoot a problem with cryptocurrency mining, entirely unrelated to this), there were no logitech drivers for linux and I was unable to set the correct DPI and macros.

So I booted back into windows and switched to the onboard memory mode and setup my profiles on that, which made them accessible to me when not using a windows environment. I noticed that after doing this, I was no longer having any problems with this player not recognizing my media key inputs. (on both the linux and windows versions of this program.)

I've attached some images of the driver to better show what I'm referring to. The first image is just of the switch between on-board and game detect mode, the second one is how you configure DPI and mouse functions when in on-board mode, the third is where you can select ingame commands or regular mouse/media functions to bind to mouse buttons when in game detect mode.

![2017-06-12_17-48-34_lcore](https://user-images.githubusercontent.com/28772701/27056794-9374e06a-4f97-11e7-8d32-cf8c31f566b0.png)
![2017-06-12_17-48-46_lcore](https://user-images.githubusercontent.com/28772701/27056795-938073f8-4f97-11e7-99a3-d5e372d8cf32.png)
![2017-06-12_17-49-45_lcore](https://user-images.githubusercontent.com/28772701/27056796-93820736-4f97-11e7-8d33-2135b7413e0f.png)

**Comment by infurnus** [comment]
I can confirm the Logitech Gaming Software is culprit for broken/conflicting media keys. I have the G700 Logitech mouse. I have play/pause and next keybound to mb5/mb6. Had for a while (2+ years), with no issues, recently reinstalled windows and this all the sudden became a problem. I've always use the on-board memory over game detect, but that isn't the default setting. @aahrg solution is very elegant, but mine already had the correct binding for mb5/mb6/mb7. I created new profiles, but it didn't change. Volume controls work fine, but play/pause and next do not. If I exit the Logitech Gaming Software, or change "Automatic Game Detection" to "On-Board Memory" on the mouses homepage, everything works as expected. I also have the Logitech Orion Spark (keyboard) and the hardware media controls work fine on it, even while the mouse media keybinds are broke. I don't know what to make of that.

Logitech has a test window within the mouse profiles that shows button events. When I push the buttons assigned to the media controls for play/pause and next, they do not fire. If I hit mute, or vol up/vol down, those trigger on depress and release in the logs. I'm not sure why I wouldn't see the triggers in the logitech software, because I don't have any keybinds directly in GPMDP or elsewhere, but just something odd I noticed. I don't have any chrome plugins (removed media keys and GMP), and rebooted several times.

@aahrg thanks again for following up on your post, both your question, and answer, were exactly what I needed to see.

**Comment by MarshallOfSound** [comment]
Logitech has always been a nightmare with their keyboard / mouse software.  As far as I know there are workarounds and apparently one is in this thread ^^.  I can't support every mouse / keyboard known to man especially when manufacturers do weird stuff and don't conform to the microsoft spec.

TLDR: Try all the workarounds, one of them will work 👍
